superset/async_events/async_query_manager.py

```diff
@@ -191,20 +191,40 @@ def submit_explore_json_job(
         force: Optional[bool] = False,
         user_id: Optional[int] = None,
     ) -> dict[str, Any]:
+        # pylint: disable=import-outside-toplevel
+        from superset import security_manager
+
         job_metadata = self.init_job(channel_id, user_id)
         self._load_explore_json_into_cache_job.delay(
-            job_metadata,
+            {**job_metadata, "guest_token": guest_user.guest_token}
+            if (guest_user := security_manager.get_current_guest_user_if_guest())
+            else job_metadata,
```

Comment:
As we have the same logic in two places, could we DRY this out into a simple local util `_get_job_metadata(job_metadata)` to avoid divergence down the line?